tests/functional/test_endpoints_v2.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2012-2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import json
+import os
+
+import pytest
+
+from botocore.endpoints_v2 import (
+    EndpointProvider,
+    EndpointResolutionError,
+    RuleSetEndpoint,
+)
+from botocore.loaders import Loader
+
+
+@pytest.fixture
+def partitions():
+    loader = Loader()
+    return loader.load_data('partitions')
```

Comment:
Because the (default) scope of this fixture is each individual test, and because you create a new `Loader` instance inside the fixture, the `partitions.json` file gets read for every test, i.e. 384 times.

There are two workarounds: 

1: Use `@pytest.fixture(scope='module')` to only evaluate the fixture once. The downside is that any test could modify the value of `partitions` and subsequent tests would get the modified value.

2: Somehow cache the loader object, maybe by moving it outside the fixture. The Loader class has caching built-in.

```python
loader = Loader()

@pytest.fixture
def partitions():
    return loader.load_data('partitions')
```

3: A combination of the previous two:

```python
@pytest.fixture(scope='module')
def loader():
    return Loader()

@pytest.fixture
def partitions(loader):
     return loader.load_data('partitions')
```

I tried option 1 and it sped up `pytest tests/functiona/test_endpoints_v2.py` by approximately 10-15%> Not a lot, but not nothing either.

Reply:
in the spirit of sticking with pytest as much as possible, i think i'll go with option 2. but yes we definitely should only load the partitions file once

Reply:
Does 2 actually solve the issue? While we're no longer creating the loader, we're presumably still loading the partitions in every test. I'd think we'd want the partitions constant at the module level.

Reply:
it actually ended up making more sense to do a module level constant since a non-test function `ruleset_testcases` uses it.

Reply:
@nateprewitt 2 solves the problem because the `Loader.load_data` method is decorated with `@instance_cache`. Repeated calls with the same filename will hit the cache.